### PR TITLE
ACMS-1500: Added modules uninstall validator for site studio packages.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1500

**Proposed changes**
Validate site studio packages deleted before uninstalling the module.

**Alternatives considered**
NA

**Testing steps**
- Install site with this code
- Install site studio module import packages
- Now try to uninstall any module having site studio package.
- It should give warning.
- Verify site.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_common/src/AcmsModulesUninstallValidator.php [hunk 1]
```diff
@@ -140,4 +172,70 @@ protected function hasMedia(string $media_type): bool {
...
+        case 'acquia_cms_site_studio':
+          $packages = ['pack_acquia_cms_core'];
+          break;
+
+        case 'acquia_cms_video':
+          $packages = ['pack_acquia_cms_video'];
+          break;
+      }
+      foreach ($packages as $package) {
+        $config = \Drupal::configFactory()->getEditable($package);
```
Comment (deepakmishra2) [inline modules/acquia_cms_common/src/AcmsModulesUninstallValidator.php:234]: Can we use dependency injection here?

**Review by deepakmishra2 — CHANGES_REQUESTED** [review]
@rajeshreeputra please have a look at the comment.

File: modules/acquia_cms_common/src/AcmsModulesUninstallValidator.php (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/AcmsModulesUninstallValidator.php]: done

**Review by deepakmishra2 — APPROVED** [review]
Looks good to me

**Review by apathak18 — APPROVED** [review]
Looks good to me

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**